src/cpp/fastdds/domain/DomainParticipantImpl.cpp

```diff
@@ -2114,6 +2114,25 @@ bool DomainParticipantImpl::has_active_entities()
     return false;
 }
 
+ReturnCode_t DomainParticipantImpl::enable_monitor_service() const
+{
+    return rtps_participant_->enable_monitor_service();
```

Comment:
This is already implementation. As said in another suggestion, RETCODE_UNSUPPORTED is not defined in the RTPS layer, so it should be returned from here.